ably/realtime/connection.py

```diff
@@ -348,13 +364,24 @@ def on_connected(self, connection_details: ConnectionDetails):
         if self.__ttl_task:
             self.__ttl_task.cancel()
         self.__connection_details = connection_details  # RTN21
+        if self.__connection_host != self.options.get_realtime_host():  # RTN17e
+            self.options.fallback_realtime_host = self.__connection_host
         if self.__state == ConnectionState.CONNECTED:  # RTN24
             state_change = ConnectionStateChange(ConnectionState.CONNECTED, ConnectionState.CONNECTED,
                                                  ConnectionEvent.UPDATE)
             self._emit(ConnectionEvent.UPDATE, state_change)
         else:
             self.enact_state_change(ConnectionState.CONNECTED)
 
+    def on_disconnected(self, msg: dict):
+        self.enact_state_change(ConnectionState.DISCONNECTED)
```

Comment:
You'll need to set the `reason` kwarg if you're transitioning the state to `DISCONNECTED`